### PR TITLE
fix(collaborator):updated permission checking on file upload/update/d…

### LINKED ISSUE #573: collaborator should be able to update the file
collaborator should be able to 
- update file

**NB:** owner pays for it

collaborator not able to
- delete file

related system tests should be corrected on 

discussion on https://0chain.slack.com/archives/C02AV6MKT36/p1646473501794929?thread_ts=1646472281.781539&cid=C02AV6MKT36

### PR DESCRIPTION
**Pull request by cnlangzi**
### Changes
-

### Fixes
- https://github.com/0chain/blobber/issues/573

### Tests
Tasks to complete before merging PR:
- [ ]  Ensure system tests are passing. If not [Run them manually](https://github.com/0chain/blobber/actions/workflows/system_tests.yml) to check for any regressions :clipboard:
- [ ]  Do any new system tests need added to test this change? do any existing system tests need updated? If so create a PR at [0chain/system_test](https://github.com/0chain/system_test)
- [ ]  Merge your system tests PR to master AFTER merging this PR

### Associated PRs (Link as appropriate):
-system_test: https://github.com/0chain/system_test/pull/225

### CODE REVIEW

File: code/go/0chain.net/blobbercore/handler/storage_handler.go [hunk 1]
```diff
@@ -527,22 +527,32 @@ func (fsh *StorageHandler) getReferencePath(ctx context.Context, r *http.Request
...
+	// it is not owner, check if it is a colloborator
+	if allocationObj.OwnerID != clientID && len(clientID) > 0 {
+		if reference.IsCollaboratorInAllPaths(ctx, allocationID, paths, clientID) {
+			publicKey = ctx.Value(constants.ContextKeyClientKey).(string)
+		}
+	}
+
+	valid, err := verifySignatureFromRequest(allocationTx, clientSign, publicKey)
+	if !valid || err != nil {
+		errCh <- common.NewError("invalid_signature", "Invalid signature")
```
Comment (peterlimg) [inline code/go/0chain.net/blobbercore/handler/storage_handler.go]: Perhaps return error of `common.NewError("invalid_signature", "could not verify the allocation owner or colloborator") to make the error message more obvious.

**Review by peterlimg — CHANGES_REQUESTED** [review]

File: code/go/0chain.net/blobbercore/handler/storage_handler.go (hunk 1 above)
Comment (cnlangzi) [reply code/go/0chain.net/blobbercore/handler/storage_handler.go]: fixed

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | fix/collaborator_update  |
| gosdk  | NOT OVERRIDDEN  |
| zboxcli  | staging  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | fix/collaborator_update  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [success] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | fix/collaborator_update  |
| gosdk  | NOT OVERRIDDEN  |
| zboxcli  | staging  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | fix/collaborator_update  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Review by peterlimg — APPROVED** [review]
lgtm

**Pull request merged**